lib/spack/spack/cmd/ci.py

```diff
@@ -76,23 +76,40 @@ def setup_parser(subparser):
         default=False,
         help="(DEPRECATED) disable DAG scheduling (use 'plain' dependencies)",
     )
-    prune_group = generate.add_mutually_exclusive_group()
-    prune_group.add_argument(
+    prune_dag_group = generate.add_mutually_exclusive_group()
+    prune_dag_group.add_argument(
         "--prune-dag",
         action="store_true",
         dest="prune_dag",
         default=True,
         help="skip up-to-date specs\n\n"
         "do not generate jobs for specs that are up-to-date on the mirror",
     )
-    prune_group.add_argument(
+    prune_dag_group.add_argument(
         "--no-prune-dag",
         action="store_false",
         dest="prune_dag",
         default=True,
         help="process up-to-date specs\n\n"
         "generate jobs for specs even when they are up-to-date on the mirror",
     )
+    prune_ext_group = generate.add_mutually_exclusive_group()
```

Comment:
Would this make sense as an accumulating enum type arg instead?

```
spack ci generate --prune-dag up-to-date --prune-dag external --prune-dag untouched ...
```

Then we could add untouched as a CLI or environment variable option, environment variable overrides?

We can also address this later, I don't think bike shedding this needs to block the refactor.

Reply:
Yeah maybe we can save it for a later discussion?